### PR TITLE
Extend naming checkers

### LINKED ISSUE #140: Bug: Some commented lines are parsed as test case/keyword
This
```
*** Keywords ***
Keyword
   Some Keyword
   Log  1

 ### I am comment ###
```
Parses to two keyword definitions: "Keyword" and " ### I am comment ###"

### LINKED ISSUE #145: Rule: Keyword name is reserved word
This rule could be breached while defining new keyword/using it:
```
*** Keywords ***
FOR
    Pass
```
Or as a typo when trying to use Robot Framework syntax:
```
FOr  ${i}  IN  ${elems}
    Log  ${i}
end
```
'FOr' and 'end' will be interpreted as two keywords "FOr" and "end". During test execution it will throw an error (with the exception for this bug):
https://github.com/robotframework/robotframework/issues/3648

### LINKED ISSUE #151: Support for various ways of parsing settings header
From RF user guide:

> Different sections are recognized by their header row. The recommended header format is *** Settings ***, but the header is case-insensitive, surrounding spaces are optional, and the number of asterisk characters can vary as long as there is one asterisk in the beginning. In addition to using the plural format, also singular variants like Setting and Test Case are accepted. In other words, also *setting would be recognized as a section header.

We need rules to ensure *** Settings ***, *** Test Case *** and so on is used, not *setting ;)

### LINKED ISSUE #152: Rule: Nested for loops are not supported
Nested for loop are not supported. You can only call keyword with another for loop inside your for loop.

### LINKED ISSUE #153: Rule: Setting name should be capitalized
Settings names like Suite Setup or Library are case insensitive. It should be reported as warning if used as not capitalized or upper case:
```
*** Settings ***
Suite Setup  Keyword
suite Setup  Keyword
SUITE SETUP  Keyword
```

### PR DESCRIPTION
**Pull request by bhirsz**
Closes #151 
Closes #153 
Closes #145 
Closes #140 
Partly relates to #150

### CODE REVIEW

**Comment by bhirsz** [comment]
Closes #145

File: robocop/checkers/naming.py [hunk 1]
```diff
@@ -58,48 +61,197 @@ def visit_KeywordName(self, node):  # noqa
...
+    """ Checker for keyword naming violations. """
     rules = {
         "0302": (
             "not-capitalized-keyword-name",
             "Keyword name should be capitalized",
             RuleSeverity.WARNING
+        ),
+        "0303": (
+            "keyword-name-is-reserved-word",
+            "'%s' is a reserved keyword%s",
```
Comment (mnojek) [inline robocop/checkers/naming.py:74]: Shouldn't there be a space before injected second string?

File: robocop/checkers/naming.py [hunk 2]
```diff
@@ -58,48 +61,197 @@ def visit_KeywordName(self, node):  # noqa
...
+                    )
+        elif self.check_if_keyword_is_reserved(keyword_name, node):
+            return
         words = keyword_name.replace('_', ' ').split(' ')
-        if any(not word.istitle() for word in words):
+        if any(not (word.istitle() or word.isupper()) for word in words if not self.is_variable(word)):
             self.report("not-capitalized-keyword-name", node=node)
+
+    def is_variable(self, name):
+        if len(name) < 3:
```
Comment (mnojek) [inline robocop/checkers/naming.py:167]: I think it can be even `< 4`

File: robocop/checkers/naming.py (hunk 1 above)
Comment (mnojek) [inline robocop/checkers/naming.py:74]: Lacking space before injected str?

File: robocop/checkers/naming.py [hunk 3]
```diff
@@ -58,48 +61,197 @@ def visit_KeywordName(self, node):  # noqa
...
+            return False
+        reserved_type = self.reserved_words[keyword_name.lower()]
+        suffix = self.prepare_reserved_word_rule_message(keyword_name, reserved_type)
+        self.report("keyword-name-is-reserved-word", keyword_name, suffix, node=node)
+        return True
+
+    @staticmethod
+    def prepare_reserved_word_rule_message(name, reserved_type):
+        return f". It must be in uppercase ({name.upper()}) when used as a marker with '{reserved_type}'. " \
+               f"Each marker should have minimum of 2 spaces as seperator." if reserved_type else ''
```
Comment (mnojek) [inline robocop/checkers/naming.py:182]:
```suggestion
               f"Each marker should have minimum of 2 spaces as separator." if reserved_type else ''
```

File: robocop/checkers/naming.py [hunk 4]
```diff
@@ -58,48 +61,197 @@ def visit_KeywordName(self, node):  # noqa
...
+class SettingsNamingChecker(VisitorChecker):
+    rules = {
+        "0306": (
+            "setting-name-not-capitalized",
+            "Setting name should be capitalized or upper case",
+            RuleSeverity.WARNING
+        ),
+        "0307": (
+            "section-name-invalid",
+            "Section name should should be in format *** Capitalized *** or *** UPPERCASE ***",
```
Comment (mnojek) [inline robocop/checkers/naming.py:194]:
```suggestion
            "Section name should should be in format '*** Capitalized ***' or '*** UPPERCASE ***'",
```

File: robocop/checkers/naming.py [hunk 5]
```diff
@@ -58,48 +61,197 @@ def visit_KeywordName(self, node):  # noqa
...
+        ),
+        "0304": (
+            "not-enough-whitespace-after-newline-marker",
+            "Provide at least two spaces after '...' marker",
+            RuleSeverity.ERROR
+        ),
+        "0305": (
+            "invalid-comment",
+            "Invalid comment. '#' needs to be first character in the cell. "
+            "For block comments you can use *** Comments *** section",
```
Comment (mnojek) [inline robocop/checkers/naming.py:85]:
```suggestion
            "For block comments you can use '*** Comments ***' section",
```

File: robocop/checkers/naming.py (hunk 1 above)
Comment (bhirsz) [reply robocop/checkers/naming.py:74]: Last injected string can be empty - for 'while' and 'continue' reserved words (which don't have 'reserved_type' like for -> For loop, else -> Run keyword if etc).

File: robocop/checkers/naming.py (hunk 2 above)
Comment (bhirsz) [reply robocop/checkers/naming.py:167]: You're right, ${a} is already 4, and ${} is not variable (probably, I will test ;))

**Review by mnojek — APPROVED** [review]

**Pull request merged**